### DISCUSSION TITLE
【第 24 周】os-checker 发布 v0.5.0；一些琐事和未完成的事情；年终总结/展望

### DISCUSSION THREAD

**Discussion by zjp-CN** [os-checker/os-checker#249] [Announcements]
## os-checker 发布 v0.5.0

* https://github.com/os-checker/os-checker/pull/248
  * https://github.com/os-checker/os-checker/pull/234
  * https://github.com/os-checker/os-checker/pull/239
  * https://github.com/os-checker/os-checker/pull/244
  * https://github.com/os-checker/os-checker/pull/231
  * https://github.com/os-checker/os-checker/pull/233
  * https://github.com/os-checker/os-checker/pull/236
  * https://github.com/os-checker/os-checker/pull/238
  * https://github.com/os-checker/os-checker/pull/247

## 向 osl 组件库提交修复和报告

用于解决 https://github.com/os-checker/os-checker/issues/219 的一部分： osl 代码无法在最新的编译器上编译，osl 及其依赖需要一些修复才能编译

* https://github.com/guoweikang/osl/issues/3
* https://github.com/guoweikang/osl/pull/2
* https://github.com/Starry-OS/taskctx/pull/9

~~目前暂未收到作者的回复或合并。~~ 提交的修复已被合并。

## 琐事

* 使用由 os-checker 维护的 outdated 检查工具，以支持 Rust Edition 2024 代码
  * https://github.com/os-checker/os-checker/pull/229
  * https://github.com/os-checker/dockerfiles/pull/3
* 注意到一些的新诊断，继续记录和分析它们
  * https://github.com/os-checker/os-checker/issues/227
  * https://github.com/os-checker/os-checker/issues/228
  * https://github.com/os-checker/os-checker/issues/232
  * https://github.com/os-checker/os-checker/issues/243
  * https://github.com/os-checker/os-checker/issues/245
* 在 RAP 仓库设置脚本维护已支持的测例，如果新的代码破坏这些测例，CI 不会通过
  * https://github.com/Artisan-Lab/RAP/pull/85
  * https://github.com/Artisan-Lab/RAP/pull/86
  * https://github.com/Artisan-Lab/RAP/pull/87
* WebUI： https://github.com/os-checker/os-checker.github.io/pull/136
* https://github.com/os-checker/plugin-cargo/pull/24
* kern-crates 更新《模块发布要求》文档： https://github.com/kern-crates/.github/pull/25

## 未完成的事情

* 试图修复 geiger 检查工具以支持 Rust Edition 2024 代码，记录见 https://github.com/os-checker/os-checker/issues/226 。
* 在徐老师的这边，为了支持 RAP 验证标准库，除了需要 `RUSTFLAGS="-Zalways-encode-mir" cargo build -Zbuild-std --target x86_64-unknown-linux-gnu` 编译标准库来获取 MIR，还需要找到 Rust 编译器 API 接口来在编译标准库期间查询标准库生成的所有可用的 MIR（可能需要查看 Miri 或者 Kani 的一些代码）。[与编译器成员 bjorn3 的交流](https://rust-lang.zulipchat.com/#narrow/channel/246057-t-cargo/topic/Callback.20analysis.20for.20Rust.20std-libs.20in.20custom.20toolchain)。
* 本来想向石磊老师的代码库提交一些修复 （背景见 https://github.com/os-checker/os-checker/issues/245 ），这些组件库自身由于各种原因而编译失败，但完全修复它们需要石老师的支持（因为他的库大多都依赖自己维护的分叉组件库），我不知道如何取得进展（虽然我知道如何修复其中一些代码）。

## 代码统计

本周 [新创建 15 个 issues 和提交 19 个 PRs](https://github.com/search?q=author%3Azjp-CN+created%3A2024-12-22T23%3A00%3A00%2B08%3A00..2024-12-29T18%3A00%3A00%2B08%3A00&type=issues)，关闭了 [16](https://github.com/search?q=author%3Azjp-CN+closed%3A2024-12-22T23%3A00%3A00%2B08%3A00..2024-12-29T18%3A00%3A00%2B08%3A00&type=issues)  个 issues（含历史）。

<details><summary>新增 400+ 行 Rust 代码。</summary>
<p>

```rust
os-checker $ git diff --stat 0ce8b4b e2b7692 "*.rs"
 os-checker-database/src/file_tree/mod.rs               |  4 +-
 os-checker-database/src/stats/mod.rs                   | 12 +++--
 os-checker-database/src/utils.rs                       |  2 +-
 os-checker-types/src/cache.rs                          | 51 ++++++++++++++++++++-
 os-checker-types/src/config.rs                         | 18 ++++++++
 os-checker-types/src/layout.rs                         | 14 +++---
 src/cli/mod.rs                                         |  6 ++-
 src/config/cmd.rs                                      | 58 ++++++++++++++++++------
 src/config/cmd/tests.rs                                |  1 +
 src/config/deserialization.rs                          | 10 +++-
 src/config/deserialization/config_options.rs           | 45 ++++++++++++++++++
 .../deserialization/config_options/type_conversion.rs  | 42 +++++++++++++++--
 src/config/deserialization/type_conversion.rs          |  4 ++
 src/config/mod.rs                                      |  2 +-
 src/config/resolve.rs                                  | 14 +++---
 src/db/cache/mod.rs                                    |  3 ++
 src/db/cache/type_conversion.rs                        |  4 ++
 src/layout/detect_targets.rs                           |  1 +
 src/layout/mod.rs                                      | 35 ++++++++++----
 src/layout/targets/mod.rs                              | 45 +++++++++++++++---
 src/layout/targets/type_conversion.rs                  | 10 ++--
 src/run_checker/mod.rs                                 | 25 ++++++----
 src/run_checker/packages_outputs.rs                    | 21 ++++++---
 src/run_checker/utils.rs                               | 29 +++++++++---
 src/utils/installation.rs                              | 20 ++++----
 src/utils/mod.rs                                       |  5 +-
 26 files changed, 390 insertions(+), 91 deletions(-)
```

```rust
plugin-cargo $ git diff --stat 3b8766c 27d3341 "*.rs"
 src/repo/mod.rs        | 31 +++++++++++++++++++++----------
 src/repo/os_checker.rs |  8 ++++++--
 2 files changed, 27 insertions(+), 12 deletions(-)
```

</p>
</details> 

## 年终总结/展望

这是 2024 年最后一周，也是 os-checker 开发的第 24 周。

最近我一直在思考如何做这个年终总结。这一路有很多很多话想说，但是我无法把它们流畅地组织起来，那就不说了。

新的一年，我希望在组件化操作系统上做更多的事情。这是一个美好的项目，感恩相遇。

**Comment by xyongcn** [comment]
持续半年的静态分析平台搭建工作，结果超出我在开头的预期。干得很漂亮👍
